### PR TITLE
exterbalance use badger instead of inmemory cache

### PR DESCRIPTION
**Pull request by niktrix**
## Problem

Asana Link: https://app.asana.com/0/inbox/1120721568345001/1124573496728450/1124573481108655

## Testing Done and Results

Created two accounts 
HDkKR1mBudicd1iprNYaxeZVn5CwqqYaEC
HHEC1x1EfgvBpX8S1UQBsqNeV7XA2vak6e

 curl http://localhost/account/HDkKR1mBudicd1iprNYaxeZVn5CwqqYaEC
{"nonce":2,"address":"HDkKR1mBudicd1iprNYaxeZVn5CwqqYaEC","balance":6389314982882443264,"storageRoot":"A2D105B456C179EE2C316D314243DB187961E855E78A62309F7E55F1CC994BCD","publicKey":"A/2t0P01JQYK6tBzwtXTPlKB6RcJElvREprupG34EQDa","erc20Address":"0x298ADBF8C71f32232fF38F1524021E4eDD7AD1e6","EBalances":{"ETH":{"Address":"0x298ADBF8C71f32232fF38F1524021E4eDD7AD1e6","Balance":6651961878500000000,"LastBlockHeight":0,"Nonce":0}}}

with ETH assets and HER tokens from external ERC20Address

after sending ETH to another address

 curl http://localhost/account/HDkKR1mBudicd1iprNYaxeZVn5CwqqYaEC
{"nonce":3,"address":"HDkKR1mBudicd1iprNYaxeZVn5CwqqYaEC","balance":6389314982882443264,"storageRoot":"995C3671A73E48C25DF07B5A3A026E0F91ED7BFF68A742270E01F752937D6752","publicKey":"A/2t0P01JQYK6tBzwtXTPlKB6RcJElvREprupG34EQDa","erc20Address":"0x298ADBF8C71f32232fF38F1524021E4eDD7AD1e6","EBalances":{"ETH":{"Address":"0x298ADBF8C71f32232fF38F1524021E4eDD7AD1e6","Balance":6151961878500000000,"LastBlockHeight":0,"Nonce":0}}}

Restart herdius-core, to check the last balance from external, previously it was from inmemory cache balance rests to external balance
Now balances are persisting in badger  syncdb

Send External TX from HDkKR1mBudicd1iprNYaxeZVn5CwqqYaEC

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -48,7 +48,7 @@ delete-db-dirs:
 	@ rm -R ./herdius
 
 create_db_dirs:
-	@ mkdir -p ./herdius && mkdir -p ./herdius/chaindb/ && mkdir -p ./herdius/statedb/
+	@ mkdir -p ./herdius && mkdir -p ./herdius/chaindb/ && mkdir -p ./herdius/statedb/ && mkdir -p ./herdius/syncdb/
```
Comment (bittelc) [inline Makefile]: How will this affect our CICD pipeline? Will I need to jump into the EC2 instance and manually run this? No problem, if so.

File: storage/exbalance/memory.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+	"fmt"
+	"os"
+
+	cryptoAmino "github.com/herdius/herdius-core/crypto/encoding/amino"
+	"github.com/herdius/herdius-core/storage/db"
+	"github.com/spf13/viper"
+	"github.com/tendermint/go-amino"
+)
+
+var cdc = amino.NewCodec()
```
Comment (bittelc) [inline storage/exbalance/memory.go]: This is my biggest pet peeve about our code. @niktrix – Do you have any ideas about how we can get away from these package-level variables. As mentioned in the inimitable Dave Che

File: Makefile (hunk 1 above)
Comment (niktrix) [reply Makefile]: are we running create_db_dirs ? if not we have to create this directory once

File: syncer/token.go [hunk 3]
```diff
@@ -91,36 +80,36 @@ func (es *HERToken) Update() {
 				es.Account.Balance -= herBalance.Uint64()
 				// val := cache.AccountCache{
 				// 	Account:              es.Account,
-				// 	LastExtBalance:       last.(cache.AccountCache).LastExtBalance,
-				// 	CurrentExtBalance:    last.(cache.AccountCache).CurrentExtBalance,
-				// 	IsFirstEntry:         last.(cache.AccountCache).IsFirstEntry,
-				// 	IsNewAmountUpdate:    last.(cache.AccountCache).IsNewAmountUpdate,
+				// 	LastExtBalance:       last.LastExtBalance,
```
Comment (bittelc) [inline syncer/token.go]: Can we just trim all this out? Does it have a purpose?

File: syncer/eth.go [hunk 4]
```diff
@@ -50,55 +50,62 @@ func (es *EthSyncer) Update() {
...
-					es.Cache.Set(es.Account.Address, last)
+					es.ExBal.Set(es.Account.Address, last)
 					return
 				}
 			} else {
-				last = last.(cache.AccountCache).UpdateLastExtBalanceByKey(assetSymbol, es.ExtBalance)
-				last = last.(cache.AccountCache).UpdateCurrentExtBalanceByKey(assetSymbol, es.ExtBalance)
-				last = last.(cache.AccountCache).UpdateIsFirstEntryByKey(assetSymbol, true)
-				last = last.(cache.AccountCache).UpdateIsNewAmountUpdateByKey(assetSymbol, false)
+				log.Println("--- ============")
```
Comment (bittelc) [inline syncer/eth.go]: Can you include some logging output for this PR, that includes lines like this?
As we expand our product's capabilities, it's important to also strengthen our logging. Every line should be informative.

File: Makefile (hunk 1 above)
Comment (bittelc) [reply Makefile]: [This is](https://github.com/herdius/herdius-core/blob/master/deployment/start_server.sh) what we're running. If you think there's an improvement to be made, go ahead and modify the startup file.

Comment (niktrix) [reply Makefile]: initially it needs to be done manually

**Pull request merged**